### ISSUE TITLE
Empty geometries don't provide __array_interface__

### ISSUE THREAD

**Issue by snorfalorpagus** [Toblerity/Shapely#403]
As title, empty geometries don't provide `__array_interface__`. e.g.:

```
>>> Point().__array_interface__
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/snorf/Desktop/Shapely/shapely/geometry/point.py", line 112, in array_interface
    ai = self.array_interface_base
  File "/Users/snorf/Desktop/Shapely/shapely/geometry/base.py", line 299, in array_interface_base
    'data': self.ctypes,
  File "/Users/snorf/Desktop/Shapely/shapely/geometry/point.py", line 100, in ctypes
    array_type = c_double * self._ndim
TypeError: can't multiply sequence by non-int of type 'NoneType'
```

I'm not sure what the correct behaviour is, but this isn't it.

**Comment by snorfalorpagus** [comment]
What is more appropriate to represent an empty geometry?

```
>>> numpy.asarray(None).__array_interface__
{'strides': None, 'typestr': '|O', 'shape': (), 'descr': [('', '|O')], 'version': 3, 'data': (69985600, False)}
>>> numpy.array([]).__array_interface__
{'strides': None, 'typestr': '<f8', 'shape': (0,), 'descr': [('', '<f8')], 'version': 3, 'data': (69985392, False)}
>>> numpy.array([[],[]]).__array_interface__
{'strides': None, 'typestr': '<f8', 'shape': (2, 0), 'descr': [('', '<f8')], 'version': 3, 'data': (68926432, False)}

**Comment by snorfalorpagus** [comment]
I think `numpy.array([])` is my preference. An empty list is the same as you get for the coordinates of an empty geometry:

```
>>> from shapely.geometry import LineString
>>> LineString().coords # empty
[]
```

**Comment by sgillies** [comment]
@snorfalorpagus I agree on this.

**Comment by snorfalorpagus** [comment]
I had a go at implementing this, but got stuck. What should the pointer in "data" point to? The empty geometry has no coordinates, so there isn't any memory to point to. It could be allocated (as it is for non-empty geometries) but what size would it be? Should it just be None? In the numpy examples I gave I'm not really sure what it's pointing to.

**Comment by sgillies** [comment]
@snorfalorpagus Considering that we have this for, e.g., a non-empty `Point`

```python
>>> Point(0, 0).__array_interface__
{'version': 3, 'data': <shapely.geometry.point.c_double_Array_2 object at 0x10215e6a8>, 'typestr': '<f8', 'shape': (2,)}
```

 I think the proper empty geometry pointer is one to a 0-length array of doubles:

```python
>>> from ctypes import c_double
>>> class Foo:
...     __array_interface__ = {'version': 3, 'typestr': '<f8', 'shape': (0,), 'data': (c_double * 0)()}
...
>>> import numpy
>>> numpy.array(Foo())
array([], dtype=float64)
```

**Comment by snorfalorpagus** [comment]
Closed by #442.
